local-macros/src/lib.rs

```diff
@@ -117,33 +123,86 @@ pub fn ast_type(attr: TokenStream, item: TokenStream) -> TokenStream {
+        "NamedDeclarationInterface" => {
+            quote! {
+                impl crate::NamedDeclarationInterface for #ast_type_name {
+                    fn name(&self) -> ::std::rc::Rc<crate::Node> {
+                        self.#first_field_name.name()
+                    }
+
+                    fn set_name(&mut self, name: ::std::rc::Rc<crate::Node>) {
+                        self.#first_field_name.set_name(name);
+                    }
+                }
+            }
+        }
+        "HasExpressionInitializerInterface" => {
+            quote! {
+                impl crate::HasExpressionInitializerInterface for #ast_type_name {
+                    fn initializer(&self) -> ::std::option::Option<::std::rc::Rc<crate::Node>> {
+                        self.#first_field_name.initializer()
+                    }
 
+                    fn set_initializer(&mut self, initializer: ::std::rc::Rc<crate::Node>) {
+                        self.#first_field_name.set_initializer(initializer);
+                    }
+                }
+            }
+        }
+        "BindingLikeDeclarationInterface" => {
+            quote! {
+                impl crate::BindingLikeDeclarationInterface for #ast_type_name {}
+            }
+        }
+        "HasTypeInterface" => {
+            quote! {
+                impl crate::HasTypeInterface for #ast_type_name {
+                    fn type_(&self) -> ::std::option::Option<::std::rc::Rc<crate::Node>> {
+                        self._variable_like_declaration.type_()
+                    }
+
+                    fn set_type(&mut self, type_: ::std::rc::Rc<crate::Node>) {
+                        self._variable_like_declaration.set_type(type_);
+                    }
+                }
+            }
+        }
+        "VariableLikeDeclarationInterface" => {
+            quote! {
+                impl crate::VariableLikeDeclarationInterface for #ast_type_name {}
+            }
+        }
+        "LiteralLikeNodeInterface" => {
+            quote! {
+                impl crate::LiteralLikeNodeInterface for #ast_type_name {
+                    fn text(&self) -> &str {
+                        self._literal_like_node.text()
+                    }
+                }
+            }
+        }
+        _ => panic!("Unknown interface: {}", interface_name),
+    }
+}
+
+fn get_enum_interface_impl(
+    interface_name: &str,
+    variant_names: &[&Ident],
+    ast_type_name: &Ident,
+) -> TokenStream2 {
+    match interface_name {
+        "NodeInterface" => {
             quote! {
                 impl crate::NodeInterface for #ast_type_name {
                     fn node_wrapper(&self) -> ::std::rc::Rc<crate::Node> {
                         match self {
-                            #(#ast_type_name::#variant_names_13(nested) => nested.node_wrapper()),*
+                            #(#ast_type_name::#variant_names(nested) => nested.node_wrapper()),*
```

Comment:
I see now what you meant [here](https://github.com/helixbass/tsc-rust/pull/11#discussion_r774780142), I didn't run into the `#variant_names_n` issue when the type of `variant_names` was a reference